### PR TITLE
[DUOS-2558] Open and controlled facet

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2558

### Summary

Adds the open and controlled facet to the data table

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by rushtong** [comment]
RE Styling ... I think most of the style is coming from mui controls, not index or bootstrap css files. When I disable those in the console, there is no change to them. While those files are still problematic, we might want to look more closely at the defaults that mui is adding.

Screenshot 2023-08-29 at 10 29 00 AM

File: src/components/data_search/DatasetSearchTable.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
+import * as React from 'react';
+import { Button } from '@mui/material';
 import { useEffect, useState } from 'react';
-import { h, div, button } from 'react-hyperscript-helpers';
```
Comment (rushtong) [inline src/components/data_search/DatasetSearchTable.js:2]: Nice

**Review by rushtong — APPROVED** [review]
Looks good!

**Comment by hams7504** [comment]
> RE Styling ... I think most of the style is coming from mui controls, not index or bootstrap css files. When I disable those in the console, there is no change to them. While those files are still problematic, we might want to look more closely at the defaults that mui is adding.
> 
> Screenshot 2023-08-29 at 10 29 00 AM

I had a similar issue with my MUI table where I had trouble changing the font style/color for certain parts of the table. I ended up using ThemeProvider from MUI styles to help fix that styling issue. That could be worth looking into. Specifically, I used styleOverrides: https://mui.com/material-ui/customization/theme-components/#theme-style-overrides. From the website: 'The theme's styleOverrides key makes it possible to potentially change every single style injected by Material UI into the DOM.'

**Review by hams7504 — APPROVED** [review]
Looks great! 👍

**Review by rushtong — APPROVED** [review]
Still 👍🏽

**Pull request merged**